### ISSUE TITLE
Shape optimization do not modifies vertices

### ISSUE THREAD

**Issue by PabloVD** [mitsuba-renderer/mitsuba3#1632]
I'm trying a simple test of shape optimization, based on the [shape optimization tutorial](https://mitsuba.readthedocs.io/en/stable/src/inverse_rendering/shape_optimization.html).

I have a road mesh in my scene. I render it, perform a dilation of the image to create a reference image, and try to use Large Steps optimization to deform the road vertices and match the target image.

However, none of the vertices are modified. Here I show the reference image (left), the initial render (middle) and the final image after optimization (right), which looks exactly the same as the initial one.

![Image](https://github.com/user-attachments/assets/7b69cb38-df7d-4399-9c74-fc33a91edec2)

The loss just does not change notably and gets stuck in some sort of local minimum. Similar behavior with x100 higher learning rates.

![Image](https://github.com/user-attachments/assets/9480fc53-5fcc-4b67-9fae-f1873dd922ea)

Any idea of where the source of the problem could be?
Thanks in advance!

This is my code

```py
import drjit as dr
import mitsuba as mi
import matplotlib.pyplot as plt
import os
import cv2

mi.set_variant('cuda_ad_rgb', 'llvm_ad_rgb')

from mitsuba import ScalarTransform4f as T

def plot_images(images):
    fig, axs = plt.subplots(1, len(images), figsize=(20, 5))
    for i in range(len(images)):
        axs[i].imshow(mi.util.convert_to_bitmap(images[i]))
        axs[i].axis('off')
    plt.show()

scene = mi.load_file("road/scene.xml")

# Update camera transform
dist = 6000
cam_origin = [0,dist,0]
sensor = mi.load_dict({
        'type': 'perspective',
        'to_world': T().look_at(target=[0, 0, 0], origin=cam_origin, up=[0, 0, 1]),
    })
params = mi.traverse(scene)
sensorparams = mi.traverse(sensor)
params["camera.to_world"] = sensorparams["to_world"]
params.update()

init_img = mi.render(scene)

# Create reference image as a dilation of the initial render
ref_image = mi.render(scene)
kernel = cv2.getStructuringElement(cv2.MORPH_RECT, (20, 20))
ref = cv2.morphologyEx(ref_image.numpy(), cv2.MORPH_DILATE, kernel)
ref_image = mi.TensorXf(ref)

params = mi.traverse(scene)

lambda_ = 25
ls = mi.ad.LargeSteps(params['shape.vertex_positions'], params['shape.faces'], lambda_)

opt = mi.ad.Adam(lr=1e-1, uniform=True)

opt['u'] = ls.to_differential(params['shape.vertex_positions'])

iterations = 1000 if 'PYTEST_CURRENT_TEST' not in os.environ else 5

losses = []

for it in range(iterations):
    
    # Retrieve the vertex positions from the latent variable
    params['shape.vertex_positions'] = ls.from_differential(opt['u'])
    params.update()
    
    img = mi.render(scene, params, sensor=sensor, seed=it)

    img = dr.mean(img, axis=-1)
    
    # L1 Loss
    loss = dr.mean(dr.abs(img - ref_image[:,:,0]))
    
    dr.backward(loss)
    opt.step()
    
    losses.append(loss.numpy())
        
    print(f"Iteration {1+it:03d}: Loss = {loss}", end='\r')

# Update the mesh after the last iteration's gradient step
params['shape.vertex_positions'] = ls.from_differential(opt['u'])
params.update()

final_img = mi.render(scene, spp=128)
plot_images([ref_image, init_img, final_img])
```

This is my scene

```xml
<scene version="3.0.0">
	<default name="integrator" value="path" />
	<default name="max_depth" value="20" />

	<integrator type="$integrator">
		<integer name="max_depth" value="$max_depth" />
	</integrator>

	<sensor type="perspective" id="camera">
		<float name="fov" value="120" />
	</sensor>
	
	<shape type="obj" id="shape">
		<string name="filename" value="models/SM_MERGED_Roads.OBJ" />
		<transform name="to_world">
			<translate x="0" y="0" z="0"/>
			<rotate y="1" angle="-60"/>
		</transform>
	</shape>

	
    <emitter type="directional">
        <vector name="direction" value="0.1886, -0.6923, -0.6965"/>
        <float name="irradiance" value="10.0"/>
    </emitter>
</scene>
```

**Comment by PabloVD** [comment]
With 100x higher learning rate the final render is slightly different, but it does not change as expected.

![Image](https://github.com/user-attachments/assets/353ff18c-d384-45ee-843b-20a3c3afecd5)

**Comment by njroussel** [comment]
Hi @PabloVD 

Crucially, you're using the wrong integrator. It should be a `*_projective` integrator, as they are the only ones to handle moving geometry. There's some notes about that in the shape optimization tutorial, there's also a slightly longer introduction to why this is necessary in this tutorial too: https://mitsuba.readthedocs.io/en/stable/src/inverse_rendering/object_pose_estimation.html

**Comment by PabloVD** [comment]
You are right, now it works much better, thanks!
